### ISSUE TITLE
feat(oxlint-cli): add CLI flag to disable type-aware rules (eg. --no-type-aware or --type-aware=false)

### ISSUE THREAD

**Issue by rujorgensen** [oxc-project/oxc#23232]
We've enabled type aware rules in our Oxlint configuration of our mono-repo by using the root config `options.typeAware: true`-option. 

However, in order to lessen friction when committing and pushing to git, we'd like to add some lightweight checks to our pre-commit hooks, and we're thinking of running something like `oxlint --quiet` without type aware rules (which runs amazingly fast at around 1.5 seconds vs. ~30 seconds with types BTW 🌟).

I've looked through the [documentation on type aware](https://oxc.rs/docs/guide/usage/linter/type-aware), and I can't find a way to overrule the root typeAware flag of the configuration file, so I'd like to suggest adding a flag for the CLI like `--no-type-aware` or `--type-aware=false` 🚀.

**Comment by camc314** [comment]
Hi!

Thanks for opening this issue and the FR.

It's not planned at the moment, however we may ship something in future.

I would suggest using a dynamic config file:

```
import { defineConfig } from 'oxlint';

const typeAware = process.env.OXLINT_SKIP_TYPE_AWARE === 'true' || true;

export default defineCOnfig({ 
  options: {
    typeAware: typeAware
  }
});
```

Then definine, and setting the relevant env var when running in pre commit hooks.

Let me know if this meets your needs - Thanks!

**Comment by rujorgensen** [comment]
@camc314 Thanks for the suggestion, that's a great workaround 👍! I don't see any reason why that wouldn't work.

I still think the flag would be more ergonomic and explicit, but feel free to close if you're not in favour 🙂.

**Comment by camc314** [comment]
> Thanks for the suggestion, that's a great workaround 👍! I don't see any reason why that wouldn't work.

Glad to hear it!

> I still think the flag would be more ergonomic and explicit, but feel free to close if you're not in favour 🙂.

Yep - will close for now, but might reopen in Q3/Q4 after more feedback etc
